### ISSUE TITLE
Funeral (FTB Custom Event) Sort

### ISSUE THREAD

**Issue by miqrogroove** [fisharebest/webtrees#3330]
When a funeral has a more specific date than a death event, the funeral is listed before the death on the individual page.  For example:

Funeral 2 October 1900
Death October 1900

It would be nice to have a more logical result in this situation.

**Comment by fisharebest** [comment]
Funeral should sort after death.

https://github.com/fisharebest/webtrees/blob/2.0.5/app/Fact.php#L100-L101

It works in my test.

Perhaps you have other data that is confusing the sorting algorithm?

Can you use the clippings cart to create a small GEDCOM file that will reproduce the problem?

**Comment by miqrogroove** [comment]
Ah that's quite different.  FTB doesn't use `_FNRL` so instead of that, mine all look like

```
1 EVEN
2 TYPE Funeral
```

For me to convert these would not be a simple substitution because they have various record IDs sprinkled between the lines.

What would be the best way to support the FTB data?

**Comment by fisharebest** [comment]
`1 EVEN/2 TYPE Funeral` is valid GEDCOM.

`1 _FNRL` is not.

I would always recommend using standard GEDCOM tags.

> What would be the best way to support the FTB data?

Your problem is not the choice of tags, but the lack of a `2 DATE`.

If there was a date, then it would sort correctly.  If you know the date of death, you can use `AFT xyz` (which sorts immediately after `xyz`)

**Comment by miqrogroove** [comment]
But then we're back at the original problem.  It's sorting the nonstandard version correctly and with the standard version it's putting an October 2nd funeral before an October death.  Is that a bug?  Or should I change the death date to BEF 2 OCT?

**Comment by fisharebest** [comment]
>  It's sorting the nonstandard version correctly

Yes - we have lots of code to do "sensible" things with third-party custom tags.

But we cannot sort events where the "type" is a free-text description that could be written in any language...

> Or should I change the death date to BEF 2 OCT?

`BEF`/`AFT` dates work well.

BTW - every funeral that I have attended has been part of either a `BURI` or `CREM` event.
These are both standard GEDCOM events.

**Comment by miqrogroove** [comment]
ok not a bug here.

**Comment by miqrogroove** [comment]
Sorry, need to reopen this.

I have an individual with 

```
1 DEAT
2 DATE 27 APR 1949
1 EVEN
2 TYPE Funeral
2 DATE 30 APR 1949
```

where it is still displaying the Funeral before the Death on that individual's page.

**Comment by miqrogroove** [comment]
Attaching a relevant screen shot.  I'm confident I will be able to edit this individual or his family to make it look right, but I'm thinking there might be a bug in here too.

![funeral](https://user-images.githubusercontent.com/1371835/84524074-5a234c00-aca7-11ea-86b9-491874ef456c.jpg)

**Comment by miqrogroove** [comment]
Yep, as soon as I deleted the Family Address fact, the display re-sorted as expected.

So is this a bug now?

**Comment by fisharebest** [comment]
What is "Family address"?

Does it come from the individual record or a family record?

**Comment by miqrogroove** [comment]
Yes it's a family record.  For reference, the change log shows record F626 with this deletion:

```
1 EVEN
2 _UID 6DC3B9AA-C957-4E78-BC92-F2BF1AAF1BD2
2 RIN MH:FF1558
2 TYPE Family Address
2 DATE 1949
2 PLAC 310 S Center St, Northville, Michigan
```

**Comment by fisharebest** [comment]
Perhaps some other facts/dates in your data might be affecting the sort algorithm.

<img width="543" alt="Screenshot 2020-06-12 at 17 26 27" src="https://user-images.githubusercontent.com/2306088/84524467-ef3b3a80-acd1-11ea-8795-9d66146efc60.png">
<img width="579" alt="Screenshot 2020-06-12 at 17 26 42" src="https://user-images.githubusercontent.com/2306088/84524475-f19d9480-acd1-11ea-8a63-c2d36f27161c.png">

Can you use the clippings cart to create a reproducable test case.

**Comment by fisharebest** [comment]
Sorry - misread your post.  Let me try again with a family event

**Comment by fisharebest** [comment]
Nope.  Still sorts OK.

**Comment by miqrogroove** [comment]
It's never simple.  I will test more.

**Comment by miqrogroove** [comment]
I couldn't find any steps that would replicate the problem.  I will consider it closed for now.
